File: Data Structures & Algorithms/is-palindrome/submission-1.js
class Solution {
    /**
     * @param {string} s
     * @return {boolean}
     */
    isPalindrome(s) {
        s = s.toLowerCase().replace(/[^a-z0-9g]/g,"");
        // two pointer solution, left and right pointers
        let left = 0;
        let right = s.length - 1;
        while(left < right){
            if(s[left] === s[right]){
                left++;
                right--;
            }else{
                return false;
            }
        }
        return true;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # so looking at this, what we are returning is a max profit
        # so we need to be tracking a min day and a current profit and get the max of a profit
        # so  we are going to be implementing a two pointer solution to keep track of our variables
        # so we cannot sort the array, prices are tied to indices (day)
        left = 0
        maxProfit = 0
        for i in range(1,len(prices)):
            if prices[left] > prices[i]:
                prices[left] = prices[i] 
            profit = prices[i] - prices[left]
            maxProfit = max(profit, maxProfit)
        return maxProfit